internal/tsoptions/parsedcommandline.go

```diff
@@ -45,3 +63,32 @@ func (p *ParsedCommandLine) GetConfigFileParsingDiagnostics() []*ast.Diagnostic
 	}
 	return p.Errors
 }
+
+func (p *ParsedCommandLine) MatchesFileName(fileName string, comparePathsOptions tspath.ComparePathsOptions) bool {
+	path := tspath.ToPath(fileName, comparePathsOptions.CurrentDirectory, comparePathsOptions.UseCaseSensitiveFileNames)
+	if slices.ContainsFunc(p.FileNames(), func(f string) bool {
+		return path == tspath.ToPath(f, comparePathsOptions.CurrentDirectory, comparePathsOptions.UseCaseSensitiveFileNames)
+	}) {
+		return true
+	}
+
+	if p.ConfigFile == nil {
+		return false
+	}
+
+	if slices.ContainsFunc(p.ConfigFile.configFileSpecs.validatedFilesSpec, func(f string) bool {
```

Comment:
You also want to check if extension is valid to avoid matching js files when they may not be included https://github.com/microsoft/TypeScript/blob/main/src/server/editorServices.ts#L4494

Reply:
I didn’t port this line because I couldn’t understand the intent. It makes `isMatchedByConfig` return `false` if the file name is supported. So only _unsupported file names_ can go on to be checked against the include and exclude specs, which doesn’t make any sense to me.

Reply:
Updated with tests adapted from matchFiles.ts